core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -1003,8 +1003,12 @@ private static void tryWriteTempFile(Path path) throws IOException {
         if (Files.exists(path)) {
             Path resolve = path.resolve(".es_temp_file");
             try {
-                Files.createFile(resolve);
-                Files.deleteIfExists(resolve);
```

Comment:
I don't think a `finally` block is enough though. If we crash hard between the creation and destruction of the file the finally block won't do anything. Say we lose power or something.

Sure, these are exceptional situations, but there isn't really any reason why we can't recover in this case and move forward. Not recovering here means that you have to read elasticsearch's code to figure out to delete the temp file.